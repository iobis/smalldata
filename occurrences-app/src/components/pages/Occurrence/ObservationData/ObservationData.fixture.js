import ObservationData from './ObservationData'

export default [{
  component: ObservationData,
  name:      'default',
  props:     {
    observationData: {
      institutionCode:         '',
      collectionCode:          '',
      fieldNumber:             '',
      catalogNumber:           '',
      recordNumber:            '',
      identifiedBy:            [],
      recordedBy:              [],
      references:              []
    },
    onChange:        console.log
  }
}]
