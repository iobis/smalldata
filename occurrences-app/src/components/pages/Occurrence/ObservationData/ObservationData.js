import CopyPreviousData from '../CopyPreviousData'
import InputText from '../../../form/InputText'
import Textarea from '../../../form/Textarea'
import PropTypes from 'prop-types'
import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'

export default function ObservationData({ onChange }) {
  const { t } = useTranslation()
  const [identifiedByNames, setIdentifiedByNames] = useState(['name 1', 'name 2'])
  const [identifiedByBy, setIdentifiedBy] = useState('')
  const [recordedByNames, setRecordedByNames] = useState(['name 1', 'name 2', 'name 3'])
  const [recordedBy, setRecordedBy] = useState('')
  const [references, setReferences] = useState(['https://google.com', 'https://gmail.com'])
  const [reference, setReference] = useState('')

  return (
    <div className="observation-data section is-fluid">
      <div className="columns">
        <InputText className="is-3" name="occurrenceForm.observationData.institutionCode" onChange={onChange}/>
        <InputText className="is-3" name="occurrenceForm.observationData.collectionCode" onChange={onChange}/>
      </div>
      <div className="columns">
        <InputText className="is-3" name="occurrenceForm.observationData.fieldNumber" onChange={onChange}/>
        <InputText className="is-3" name="occurrenceForm.observationData.catalogNumber" onChange={onChange}/>
        <InputText className="is-3" name="occurrenceForm.observationData.recordNumber" onChange={onChange}/>
      </div>
      <div className="columns">
        <div className="column field is-3">
          <label className="label">
            {t('occurrenceForm.observationData.identifiedBy.label')}
          </label>
          <input
            value={identifiedByBy}
            className="input"
            onChange={(e) => setIdentifiedBy(e.target.value)}
            onKeyDown={(e) => {
              if (e.key === 'Enter') {
                setIdentifiedByNames([...identifiedByNames, e.target.value])
                setIdentifiedBy('')
              }
            }}
            type="text"
            placeholder={t('occurrenceForm.observationData.identifiedBy.placeholder')}/>
          <p className="help">{t('occurrenceForm.observationData.identifiedBy.help')}</p>
          <StringTags strings={identifiedByNames} onDelete={(names) => setIdentifiedByNames(names)}/>
        </div>
        <div className="column field is-3">
          <label className="label">
            {t('occurrenceForm.observationData.recordedBy.label')}
          </label>
          <input
            value={recordedBy}
            className="input"
            onChange={(e) => setRecordedBy(e.target.value)}
            type="text"
            onKeyDown={(e) => {
              if (e.key === 'Enter') {
                setRecordedByNames([...recordedByNames, e.target.value])
                setRecordedBy('')
              }
            }}
            placeholder={t('occurrenceForm.observationData.recordedBy.placeholder')}/>
          <p className="help">{t('occurrenceForm.observationData.recordedBy.help')}</p>
          <StringTags strings={recordedByNames} onDelete={(names) => setRecordedByNames(names)}/>
        </div>
      </div>
      <div className="columns">
        <InputText
          className="is-9"
          name="occurrenceForm.observationData.identificationQualifier"
          onChange={onChange}/>
      </div>
      <div className="columns">
        <Textarea
          className="is-9"
          name="occurrenceForm.observationData.identificationRemarks"
          onChange={onChange}/>
      </div>
      <div className="columns">
        <div className="column field is-9">
          <label className="label">
            {t('occurrenceForm.observationData.references.label')}
          </label>
          <input
            value={reference}
            className="input"
            onChange={(e) => setReference(e.target.value)}
            type="text"
            onKeyDown={(e) => {
              if (e.key === 'Enter') {
                setReferences([...references, e.target.value])
                setReference('')
              }
            }}
            placeholder={t('occurrenceForm.observationData.references.placeholder')}/>
          <p className="help">{t('occurrenceForm.observationData.references.help')}</p>
          <LinkTags links={references} onDelete={(names) => setReferences(names)}/>
        </div>
      </div>
      <CopyPreviousData/>
    </div>
  )
}

ObservationData.propTypes = {
  onChange: PropTypes.func.isRequired
}

function LinkTags({ links, onDelete }) {
  return (
    <div className="block">
      {links.map((link, index) =>
        <div key={index}>
          <span className="tag">
            <a href={link}>{link}</a>
            <button className="delete is-small" onClick={() => onDelete(deleteByIndex(index))}/></span>
        </div>
      )}
    </div>
  )

  function deleteByIndex(index) {
    return [...links.slice(0, index), ...links.slice(index + 1)]
  }
}

LinkTags.propTypes = {
  links:    PropTypes.arrayOf(PropTypes.string).isRequired,
  onDelete: PropTypes.func.isRequired
}

function StringTags({ strings, onDelete }) {
  return (
    <div className="block">
      {strings.map((string, index) =>
        <div key={index}>
          <span className="tag">
            {string}
            <button className="delete is-small" onClick={() => onDelete(deleteByIndex(index))}/></span>
        </div>
      )}
    </div>
  )

  function deleteByIndex(index) {
    return [...strings.slice(0, index), ...strings.slice(index + 1)]
  }
}

StringTags.propTypes = {
  strings:  PropTypes.arrayOf(PropTypes.string).isRequired,
  onDelete: PropTypes.func.isRequired
}
