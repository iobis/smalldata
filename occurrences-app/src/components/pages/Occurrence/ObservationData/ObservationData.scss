.observation-data {
  margin-bottom: 1.25rem;
}
