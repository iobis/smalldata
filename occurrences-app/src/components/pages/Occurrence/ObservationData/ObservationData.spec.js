import React from 'react'
import renderer from 'react-test-renderer'
import ObservationData from './ObservationData'

describe('ObservationData', () => {
  it('renders correctly', () => {
    expect(renderer.create(createComponent()).toJSON()).toMatchSnapshot()
  })
})

function createComponent() {
  const observationData = {
    institutionCode:         '',
    collectionCode:          '',
    fieldNumber:             '',
    catalogNumber:           '',
    recordNumber:            '',
    identifiedBy:            ['name 1', 'name 2'],
    recordedBy:              ['name 1', 'name 2', 'name 3'],
    identificationQualifier: '',
    identificationRemarks:   '',
    references:              ['https://google.com', 'https://gmail.com']
  }
  return (
    <ObservationData observationData={observationData} onChange={jest.fn()}/>
  )
}
