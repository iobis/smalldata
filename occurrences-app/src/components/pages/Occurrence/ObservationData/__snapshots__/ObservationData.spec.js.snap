// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationData renders correctly 1`] = `
<div
  className="observation-data section is-fluid"
>
  <div
    className="columns"
  >
    <div
      className="column field is-3"
    >
      <label
        className="label"
      >
        occurrenceForm.observationData.institutionCode.label
      </label>
      <input
        className="input"
        onChange={[MockFunction]}
        type="text"
      />
    </div>
    <div
      className="column field is-3"
    >
      <label
        className="label"
      >
        occurrenceForm.observationData.collectionCode.label
      </label>
      <input
        className="input"
        onChange={[MockFunction]}
        type="text"
      />
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="column field is-3"
    >
      <label
        className="label"
      >
        occurrenceForm.observationData.fieldNumber.label
      </label>
      <input
        className="input"
        onChange={[MockFunction]}
        type="text"
      />
    </div>
    <div
      className="column field is-3"
    >
      <label
        className="label"
      >
        occurrenceForm.observationData.catalogNumber.label
      </label>
      <input
        className="input"
        onChange={[MockFunction]}
        type="text"
      />
    </div>
    <div
      className="column field is-3"
    >
      <label
        className="label"
      >
        occurrenceForm.observationData.recordNumber.label
      </label>
      <input
        className="input"
        onChange={[MockFunction]}
        type="text"
      />
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="column field is-3"
    >
      <label
        className="label"
      >
        occurrenceForm.observationData.identifiedBy.label
      </label>
      <input
        className="input"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="occurrenceForm.observationData.identifiedBy.placeholder"
        type="text"
        value=""
      />
      <p
        className="help"
      >
        occurrenceForm.observationData.identifiedBy.help
      </p>
      <div
        className="block"
      >
        <div>
          <span
            className="tag"
          >
            name 1
            <button
              className="delete is-small"
              onClick={[Function]}
            />
          </span>
        </div>
        <div>
          <span
            className="tag"
          >
            name 2
            <button
              className="delete is-small"
              onClick={[Function]}
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="column field is-3"
    >
      <label
        className="label"
      >
        occurrenceForm.observationData.recordedBy.label
      </label>
      <input
        className="input"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="occurrenceForm.observationData.recordedBy.placeholder"
        type="text"
        value=""
      />
      <p
        className="help"
      >
        occurrenceForm.observationData.recordedBy.help
      </p>
      <div
        className="block"
      >
        <div>
          <span
            className="tag"
          >
            name 1
            <button
              className="delete is-small"
              onClick={[Function]}
            />
          </span>
        </div>
        <div>
          <span
            className="tag"
          >
            name 2
            <button
              className="delete is-small"
              onClick={[Function]}
            />
          </span>
        </div>
        <div>
          <span
            className="tag"
          >
            name 3
            <button
              className="delete is-small"
              onClick={[Function]}
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="column field is-9"
    >
      <label
        className="label"
      >
        occurrenceForm.observationData.identificationQualifier.label
      </label>
      <input
        className="input"
        onChange={[MockFunction]}
        type="text"
      />
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="column field is-9"
    >
      <label
        className="label"
      >
        occurrenceForm.observationData.identificationRemarks.label
      </label>
      <textarea
        className="textarea"
        onChange={[MockFunction]}
        rows={5}
      />
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="column field is-9"
    >
      <label
        className="label"
      >
        occurrenceForm.observationData.references.label
      </label>
      <input
        className="input"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="occurrenceForm.observationData.references.placeholder"
        type="text"
        value=""
      />
      <p
        className="help"
      >
        occurrenceForm.observationData.references.help
      </p>
      <div
        className="block"
      >
        <div>
          <span
            className="tag"
          >
            <a
              href="https://google.com"
            >
              https://google.com
            </a>
            <button
              className="delete is-small"
              onClick={[Function]}
            />
          </span>
        </div>
        <div>
          <span
            className="tag"
          >
            <a
              href="https://gmail.com"
            >
              https://gmail.com
            </a>
            <button
              className="delete is-small"
              onClick={[Function]}
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="copy-previous-data column is-narrow"
    >
      copy data from previous entry
    </div>
  </div>
</div>
`;
